src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -94,6 +122,190 @@ typedef int         (*t8_forest_vtk_cell_data_kernel) (t8_forest_t forest,
+      for (ivertex = 0; ivertex < t8_eclass_num_vertices[tree->eclass];
+           ivertex++, point_id++) {
+        /* We take the element coordinates in vtk order */
+        t8_forest_element_coordinate (forest, itree, element,
+                                      vertices,
+                                      t8_eclass_vtk_corner_number
+                                      [tree->eclass]
+                                      [ivertex], coordinates);
+        x = coordinates[0];
+        y = coordinates[1];
+        z = coordinates[2];
+        /* Insert point in the points array */
+        points->InsertNextPoint (x, y, z);
+        /* depending on the type we add points to the correct element */
+        switch (element_shape) {
+        case T8_ECLASS_VERTEX:
+          vertex->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_LINE:
+          line->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_QUAD:
+          quad->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_TRIANGLE:
+          tri->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_HEX:
+          hexa->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_TET:
+          tet->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_PRISM:
+          prism->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        case T8_ECLASS_PYRAMID:
+          pyramid->GetPointIds ()->SetId (ivertex, point_id);
+          break;
+        default:
+          printf ("Kein Elementtyp.\n");
+          break;
+        }
+
+      }
+      /* depending on the element type we insert next cell in the cell array */
+      switch (element_shape) {
+      case T8_ECLASS_VERTEX:
+        cellArray->InsertNextCell (vertex);
+        break;
+      case T8_ECLASS_LINE:
+        cellArray->InsertNextCell (line);
+        break;
+      case T8_ECLASS_QUAD:
+        cellArray->InsertNextCell (quad);
+        break;
+      case T8_ECLASS_TRIANGLE:
+        cellArray->InsertNextCell (tri);
+        break;
+      case T8_ECLASS_HEX:
+        cellArray->InsertNextCell (hexa);
+        break;
+      case T8_ECLASS_TET:
+        cellArray->InsertNextCell (tet);
+        break;
+      case T8_ECLASS_PRISM:
+        cellArray->InsertNextCell (prism);
+        break;
+      case T8_ECLASS_PYRAMID:
+        cellArray->InsertNextCell (pyramid);
+        break;
+      default:
+        printf ("Kein Elementtyp.\n");
+        break;
+      }
+      cellTypes[elem_id] = t8_eclass_vtk_type[element_shape];
+      elem_id++;
+    }
+  }
+
+  // Write file
+  vtkNew < vtkUnstructuredGrid > unstructuredGrid;
+  unstructuredGrid->SetPoints (points);
+  unstructuredGrid->SetCells (cellTypes, cellArray);
+
+  std::string mpifilename = fileprefix;
+  mpifilename += ".pvtu";
```

Comment:
We avoid using std functionalities and should stick to raw c strings to be consistent with the rest of t8code.

```suggestion
  char mpifilename[BUFSIZ] = fileprefix;
  snprintf (mpifilename, BUFSIZ, "%s.pvtu", fileprefix);
```

Reply:
This gives me an error: error: array must be initialized with a brace-enclosed initializer

Reply:
ah, yes. You cannot initialize a char array with a pointer.
(i really should check the code i i suggest here).
```C
  char mpifilename[BUFSIZ];
   snprintf (mpifilename, BUFSIZ, "%s.pvtu", fileprefix);
```